### PR TITLE
fix(wallet-walletconnect): make WalletConnect `chainIds` optional

### LINKED ISSUE #399: WalletConnect `requiredNamespaces` optionnal (and so should be `chainIds`)
In dot-connect, I have to passe `chainIds` and if I omit it, or pass it an empty array instead I get an error.
> Unsupported chains. connect() requiredNamespaces, chains must be defined as "namespace:chainId" e.g. "eip155:1": {...} in the namespace key OR as an array of CAIP-2 chainIds e.g. eip155: { chains: ["eip155:1", "eip155:5"] }

In the WalletConnect types, you can see that `requiredNamespaces` and `optionalNamespaces` too. One of the 2 should definitely not be empty. Here: https://github.com/WalletConnect/walletconnect-monorepo/blob/74bb304a452725e36e0fcd70aafcb1a0f12d31ee/packages/types/src/sign-client/engine.ts#L60

From my experience, Dapps like Hydration have only passed a long list of optional namespaces, e.g https://github.com/galacticcouncil/hydration-ui/blob/master/src/sections/web3-connect/wallets/WalletConnect.ts#L274 and this is what I would like to do in Delegit too. I support many chains and don't want the wallet provider to have to support all of them. E.g Delegit supports Polkadot, Kusama, and Paseo, whereas my wallet provider would have Polkadot, Kusama, and Westend. Then just because WC requires Paseo, and the provider doesn't have it, the WC connection would fail. This is something that the wallet provider has to check, not the Dapp.

### PR DESCRIPTION
**Pull request by Tbaut**
closes #399

### CODE REVIEW

**Comment by tien** [comment]
Thanks @Tbaut, can you create a changelog by running `yarn changeset add`?

**Comment by Tbaut** [comment]
Sure, I just did it.

File: .changeset/tender-pets-warn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@reactive-dot/wallet-walletconnect": minor
+---
+
+chainIds is now optional. Either chainIds or optionalChainIds must be set.
```
Comment (tien) [inline .changeset/tender-pets-warn.md:5]:
```suggestion
`chainIds` is now optional. Either `chainIds` or `optionalChainIds` must be set.
```

File: packages/wallet-walletconnect/src/index.ts
Comment (tien) [inline packages/wallet-walletconnect/src/index.ts:1]: [This line](https://github.com/tien/reactive-dot/pull/400/files#diff-70e4d7d657aa693516150c5b63c9c0dbdb050902f7aac1c7241f62aa84248f66R61), needs to be updated to `this.#chainIds = options.chainIds ?? []` to resolves the type error.

**Review by tien — CHANGES_REQUESTED** [review]
LGTM, just a couple minor nits.

File: packages/wallet-walletconnect/src/index.ts
Comment (Tbaut) [reply packages/wallet-walletconnect/src/index.ts:1]: uh sure, I didn't see it somehow. done

**Review by tien — APPROVED** [review]

**Pull request merged**